src/llvm-gcroot.cpp

```diff
@@ -37,6 +42,74 @@ static struct {
 } jl_gc_frame_stats = {0};
 #endif
 
+typedef std::pair<CallInst*, unsigned> frame_register;
+struct liveness {
+    typedef unsigned id;
+    enum {
+        // an assignment to a gcroot exists in the basic-block
+        // (potentially no live-in from the predecessor basic-blocks)
+        assign = 1<<0,
+        // a use of a gcroot exists in the basic-block
+        // (potentially a "kill" and no live-out to the successor basic-blocks)
+        kill   = 1<<1,
+        // the gcroot is live over the entire basic-block
+        // (the assign/kill are not dominating of entry/exit)
```

Comment:
This is simply copying the original comment and I believe it means https://en.wikipedia.org/wiki/Dominator_(graph_theory). I'm not sure what's the right way to use these words in these context.